### PR TITLE
fix(language-service): reset MockHost after every spec instead of creating new LS

### PR DESCRIPTION
**Pull request by kyliau**
This commit speeds up the tests by calling `MockHost.reset()` in
    `beforeEach()` instead of destroying the entire language service and
    creating a new one. The creation of a new language service instance is
    expensive due to the need to initialize many core Symbols when creating
    a new program.
    
    This speeds ups the test (on my local machine) from 35 secs to 15 secs.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: packages/language-service/test/hover_spec.ts [hunk 1]
```diff
@@ -9,23 +9,17 @@
...
-  let ngLSHost: TypeScriptServiceHost;
-  let ngLS: LanguageService;
-
-  beforeEach(() => {
-    mockHost = new MockTypescriptHost(['/app/main.ts', '/app/parsing-cases.ts']);
-    tsLS = ts.createLanguageService(mockHost);
-    ngLSHost = new TypeScriptServiceHost(mockHost, tsLS);
-    ngLS = createLanguageService(ngLSHost);
-  });
+  const mockHost = new MockTypescriptHost(['/app/main.ts']);
```
Comment (ayazhafiz) [inline packages/language-service/test/hover_spec.ts]: Why not create the mockHost with `/app/parsing-cases.ts` as well rather than changing `StringModel` in later tests?

**Review by ayazhafiz — CHANGES_REQUESTED** [review]

File: packages/language-service/test/hover_spec.ts (hunk 1 above)
Comment (kyliau) [reply packages/language-service/test/hover_spec.ts]: The problem here is not actually the `script names` passed to `MockTypescriptHost`. 
(`main.ts` alone is enough as an entry point. Since it imports symbols from `parsing-cases.ts`, the file automatically ends up in the program.)
It's because overriding `parsing-cases.ts` leads to all sorts of undefined behavior since `main.ts` explicitly imports a set of symbols from the file.
tldr; we should never override files that export symbols different from those imported in `main.ts`. In fact, I'm thinking of removing the `override` method completely.

File: packages/language-service/test/hover_spec.ts [hunk 2]
```diff
@@ -183,20 +177,20 @@ describe('hover', () => {
   });
 
   it('should be able to find the NgModule of a directive', () => {
-    const fileName = '/app/parsing-cases.ts';
+    const fileName = '/app/app.component.ts';
     mockHost.override(fileName, `
       import {Directive} from '@angular/core';
 
       @Directive({
         selector: '[string-model]',
```
Comment (ayazhafiz) [inline packages/language-service/test/hover_spec.ts]: could you change this to be `app-component`?

**Review by ayazhafiz — CHANGES_REQUESTED** [review]

File: packages/language-service/test/hover_spec.ts (hunk 2 above)
Comment (kyliau) [reply packages/language-service/test/hover_spec.ts]: I reworked the test to use existing example in `parsing-cases.ts` instead. PTAL.

**Review by ayazhafiz — APPROVED** [review]

File: packages/language-service/test/hover_spec.ts (hunk 2 above)
Comment (ayazhafiz) [reply packages/language-service/test/hover_spec.ts]: lgtm
